spec/controllers/idv/review_controller_spec.rb

```diff
@@ -0,0 +1,148 @@
+      allow(subject).to receive(:confirm_idv_session_started).and_return(true)
+      allow(subject).to receive(:current_user).and_return(user)
+    end
+
+    context 'user has completed all steps' do
+      before do
+        allow(subject).to receive(:idv_session).and_return(params: user_attrs)
+      end
+
+      it 'shows completed session' do
+        get :new
+
+        user_attrs.values.each do |attr|
+          expect(response.body).to have_content(attr)
+        end
+      end
+    end
+
+    context 'user has missed phone step' do
+      before do
+        allow(subject).to receive(:idv_session).and_return(
+          params: user_attrs.reject { |key| key == :phone }
+        )
+      end
+
+      it 'redirects to phone step' do
+        get :new
+
+        expect(response).to redirect_to idv_phone_path
+      end
+    end
+
+    context 'user has missed finance step' do
+      before do
+        allow(subject).to receive(:idv_session).and_return(
+          params: user_attrs.reject { |key| key == :ccn }
+        )
+      end
+
+      it 'redirects to finance step' do
+        get :new
+
+        expect(response).to redirect_to idv_finance_path
+      end
+    end
+  end
+
+  describe '#create' do
+    before do
+      sign_in(user)
+      allow(subject).to receive(:confirm_two_factor_authenticated).and_return(true)
+      allow(subject).to receive(:confirm_idv_session_started).and_return(true)
+      allow(subject).to receive(:current_user).and_return(user)
+    end
+
+    context 'user has completed all steps' do
+      before do
+        allow(subject).to receive(:idv_session).and_return(
+          params: user_attrs.merge(phone_confirmed_at: Time.zone.now)
+        )
+      end
+
+      it 'redirects to questions path' do
+        put :create
+
+        expect(response).to redirect_to idv_questions_path
+      end
+    end
+
+    context 'user has entered different phone number from MFA' do
+      before do
+        allow(subject).to receive(:idv_session).and_return(
+          params: user_attrs.merge(phone: '213-555-1000')
+        )
+      end
+
+      it 'redirects to phone confirmation path' do
+        put :create
+
+        expect(response).to redirect_to idv_phone_confirmation_send_path
+      end
+    end
+
+    context 'user has missed phone step' do
+      before do
+        allow(subject).to receive(:idv_session).and_return(
+          params: user_attrs.reject { |key| key == :phone }
+        )
+      end
+
+      it 'redirects to phone step' do
+        put :create
+
+        expect(response).to redirect_to idv_phone_path
+      end
+    end
+
+    context 'user has missed finance step' do
+      before do
+        allow(subject).to receive(:idv_session).and_return(
+          params: user_attrs.reject { |key| key == :ccn }
+        )
+      end
+
+      it 'redirects to finance step' do
+        put :create
+
+        expect(response).to redirect_to idv_finance_path
+      end
+    end
```

Comment:
The duplication is that both the `new` and `create` specs are testing what happens when the user misses the phone and finance steps. Instead of repeating the test twice, we can have a single test that tests the `confirm_idv_steps_complete` method.